File: Initial practice 1200/B_Minimize_Inversions.java
import java.util.*;

public class B_Minimize_Inversions
{
    public static void main(String[] args) {
        int t=0;
        Scanner sc = new Scanner(System.in);
        Solution s = new Solution();
        t = sc.nextInt();
        sc.nextLine();
        while(t>0)
        {
            t--;
            int n=sc.nextInt();
            Pair[] list = new Pair[n];
            for(int i=0;i<n;i++){
                Pair nn = new Pair(1,2);
                nn.value=sc.nextInt();
                list[i]=nn;
            }
            for(int i=0;i<n;i++){
                list[i].index=sc.nextInt();
            }
            s.helper(list,n);
        }
        sc.close();
    }
}

class Pair{
    int value;
    int index;
    public Pair(int value,int index){
        this.value=value;
        this.index=index;
    }
}

class MyComp implements Comparator<Pair>{
    @Override
    public int compare(Pair o1, Pair o2) {
        if(o1.value<o2.value) return -1;
        else if(o1.value>o2.value) return 1;
        else return 0;
    }
}

class Solution {
    public void helper(Pair[] list,int n)
    {
        Arrays.sort(list,new MyComp());
        for(int i=0;i<n;i++){
            System.out.print(list[i].value+" ");
        }
        System.out.println();
        for(int i=0;i<n;i++){
            System.out.print(list[i].index+" ");
        }
        System.out.println();
        return;
    }
}

File: Initial practice 1200/B_Minimize_Inversions.cpp
#include <iostream>
#include<bits/stdc++.h>

using namespace std;

void helper()
{
    
}

int main()
{
    int t;
    cin>>t;
    while(t--)
    {
        int n;
        cin>>n;
        vector<pair<int,int>> list(n,make_pair(1,2));
        for(int i=0;i<n;i++){
            int val;
            cin>>val;
            list[i].first=val;
        }
        for(int i=0;i<n;i++){
            int val;
            cin>>val;
            list[i].second=val;
        }
        sort(list.begin(),list.end());
        for(int i=0;i<n;i++){
            cout<<list[i].first<<" ";
        }
        cout<<endl;
        for(int i=0;i<n;i++){
            cout<<list[i].second<<" ";           
        }
        cout<<endl;
    }
    return 0;
}